Use symlink to common

sandrobonazzola opened the pull request:
Changes proposed in this pull request:

- Using symlinks here to help moving the documentation to upstream first.

I confirm that this pull request was submitted according to the [contribution guidelines](https://github.com/oVirt/ovirt-site/blob/master/CONTRIBUTING.md): @sandrobonazzola 

This pull request needs review by: @stoobie @duck-rh

stoobie reviewed (approved):
I fetched the branch, generated ovirt-site and verified that include statements that reference the symlinked /common/ directory work as expected.
I did _not_ verify that all links to the /common/ directory were changes to reference the symlink.

duck-rh reviewed (changes requested):
I'm all in for upstream first of course \o/.

There's a regression when testing the site with `bundle exec jekyll s`:
```
        ** ERROR: directory is already being watched! **

        Directory: /home/duck/Projects/oVirt/ovirt-site/source/documentation/data_warehouse_guide/common

        is already being watched through: /home/duck/Projects/oVirt/ovirt-site/source/documentation/common
```
and others, so some patterns are not right.

sandrobonazzola commented on _config.yml:
```diff
@@ -138,27 +138,37 @@ exclude:
   - CODEOWNERS
   # avoid generating paginated asciidoc content
-  - documentation/common/**/*.adoc
+  - documentation/common/**/*
```
yes, the ** ERROR: directory is already being watched!  is due to the symlinks. the listener library doesn't like them.
Tried to silence it by adding livereload_ignore config but doesn't seem to work, so I'm open to suggestions

sandrobonazzola replied:
Addressed the comment about the patterns, added a note to the docs about the error messages due to symlinks

sandrobonazzola commented:
I addressed comments about the config file, added a  note about the symlink generated error.
I found https://github.com/guard/listen/issues/339 which explain that those errors can be avoided with monkey patching but honestly it's beyond my knowledge of the ruby language.

duck-rh commented:
@sandrobonazzola I looked into jekyll-watch and it takes the `exclude` list into account but skips them if they don't resolve outside the `source` directory and there is no way to inject custom ignores. I also discovered it does not support globbing like Jekyll does in the `ignore` field. I then dug into `listen` and despite uptream's comment on various BRs the ignore list is only skipping events on all architectures (and not just non-linux). Anyway I guess skipping non-directories would still be a problem.

Since there's no way to get the right behavior I looked in the suggested monkeypatch: I believe it is doing the right thing, skipping the warning message but failing in a way that will only skip listening to the symlink target. I have then integrated it in a new patch that I intend to push right after this message.
